assets/src/js/pdf_icon.js

```diff
@@ -1,12 +1,26 @@
 // Add pdf icon to pdf links
-export const setupPDFIcon = function($) {
-  'use strict';
+export const setupPDFIcon = () => {
+  const links = [...document.querySelectorAll('a[href*=".pdf"]')];
 
-  $('a[href*=".pdf"]').each(function() {
-    const link = $(this);
+  links.forEach(link => {
+    // We don't want to show the icon in headings/titles
+    const linkParent = link.parentElement.nodeName;
+    if (['H1', 'H2', 'H3', 'H4', 'H5', 'H6'].includes(linkParent)) {
+      return;
+    }
+
+    const linkChildren = [...link.childNodes];
+    const isImage = linkChildren.find(linkChild => linkChild.nodeName === 'IMG');
+
+    if (isImage) {
+      return;
+    } else {
+      const text = link.textContent || link.innerText;
+      if (text.trim().length === 0) {
+        return;
+      }
 
-    if (!(link.parent('h1, h2, h3, h4, h5, h6').length || link.has('img').length)) {
-      link.addClass('icon-link pdf-link');
+      link.classList.add('pdf-link');
```

Comment:
If all we're doing in the end is adding a class, maybe we could replicate the whole script in pure CSS? :thinking:

Reply:
I don't think that would be possible because we need to check that the link has no image as child, and that's not possible in pure CSS 🤔